apps/src/templates/Congrats.jsx

```diff
@@ -0,0 +1,71 @@
+import $ from 'jquery';
+import _ from 'lodash';
+import React, { PropTypes, Component } from 'react';
+import Responsive from '../responsive';
+import Certificate from './Certificate';
+import StudentsBeyondHoc from './StudentsBeyondHoc';
+import TeachersBeyondHoc from './TeachersBeyondHoc';
+
+export default class Congrats extends Component {
+  static propTypes = {
+    completedTutorialType: PropTypes.oneOf(['applab', 'newMinecraft', 'oldMinecraft', 'other']).isRequired,
+    MCShareLink: PropTypes.string,
+    isRtl: PropTypes.bool.isRequired
+  };
+
+  constructor(props) {
+    super(props);
+    this.responsive = new Responsive();
+    this.state = {
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height(),
+      mobileLayout: this.responsive.isResponsiveCategoryInactive('md')
+    };
+  }
+
+  componentDidMount() {
+    // Resize handler.
+    window.addEventListener('resize', _.debounce(this.onResize, 100).bind(this));
+  }
+
+  onResize() {
+    const windowWidth = $(window).width();
+    const windowHeight = $(window).height();
+    // We fire window resize events when the grippy is dragged so that non-React
+    // controlled components are able to rerender the editor. If width/height
+    // didn't change, we don't need to do anything else here
+    if (windowWidth === this.state.windowWidth &&
+        windowHeight === this.state.windowHeight) {
+      return;
+    }
+
+    this.setState({
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height()
+    });
+
+    this.setState({mobileLayout: this.responsive.isResponsiveCategoryInactive('md')});
```

Comment:
I believe we finally realized that changing state is simply done to trigger a re-render, and there are React APIs we could call directly instead, if we wanted.

Reply:
`.forceUpdate()`?  I think that's heavily discouraged in the React documentation. Is there another way?

Reply:
I think that's the main possibility for now.  Happy to brainstorm more later, and don't think it needs to block this work.. just wanted to call this out as we duplicate the technique.

Reply:
Let's chat about responsive React today or at standup tomorrow.  Rather than having each responsive component register it's own `onResize` handler it would be much better to register it in a single place (debounced) and put the current responsive size into the Redux store.  Then components can connect to that value and respond only when it changes.

The other issue here is the two back to back `setState` calls.

Reply:
Let's talk at standup, collectivley decide on a plan, and apply that plan here and /courses as follow up work.